auth_from_http_remote_user/tests/test_res_users.py

```diff
@@ -0,0 +1,93 @@
+##############################################################################
+#
+#    Author: Laurent Mignon
+#    Copyright 2014 'ACSONE SA/NV'
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.tests import common
+import mock
+import os
+from contextlib import contextmanager
+import unittest
+
+
+@contextmanager
+def mock_cursor(cr):
+    with mock.patch('openerp.sql_db.Connection.cursor') as mocked_cursor_call:
+        org_close = cr.close
+        org_autocommit = cr.autocommit
+        try:
+            cr.close = mock.Mock()
+            cr.autocommit = mock.Mock()
+            mocked_cursor_call.return_value = cr
+            yield
+        finally:
+            cr.close = org_close
+    cr.autocommit = org_autocommit
+
+
+class TestResUsers(common.TransactionCase):
+
+    def test_login(self):
+        self.env.cr.execute('update res_users set sso_key = null;')
+        self.env.cr.commit()
```

Comment:
is this commit really needed?

Reply:
Not sure but are we not bypassing the ORM because the authentication methods uses a different env ?

Reply:
that's not my concern :) My concern is that if you commit you are going to commit any records that might be in the pool and when you reuse a test db (like when you are in dev mode) you can have strange behaviors/errors because of persistent test records.

Reply:
Well tried removing it and the test failed.

Reply:
Isn't because of the cache? Could you try `self.env.clear()`?

Reply:
Nope tests failed as well with `self.env.clear()`
Maybe because the authentication process by pass the cache ?